compiler/prelude/jsmapping.go

```diff
@@ -132,7 +135,11 @@ var $externalize = function(v, t) {
       if (f.pkg !== "") { /* not exported */
         continue;
       }
-      o[f.name] = $externalize(v[f.prop], f.typ);
+      var name = f.name;
+      if (name === "") {
+        name = f.prop;
```

Comment:
I'm not well versed in minification, but I would have thought that strings (and all other constants) would have been left alone, especially with javascript where an object key can be referred to with and without a string. Also, would that same minification not also apply to f.name?

I do agree, though - using f.prop feels iffy, it just seemed to contain the correct value. Glancing at the gopherjs code, it seems that if the type name is a javascript keyword, it will append a $ and the struct position to the name. Not ideal, but, if reserved names aren't allowed, then there's not much alternative. Saying that, however, I was under the impression that you could use reserved words as object keys in javascript - that may need looking at.

Also, the encoding/json rules have issues. The problem is how to deal with multiple anonymous fields which share field names: -

https://play.golang.org/p/iv1x_3xVgr

...encoding/json library just removes the identically named fields, that doesn't seem appropriate for this case.